### ISSUE TITLE
`DoubleMutabilityForCollection` points to the start of kdoc on the variable

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8903]
## Expected Behavior
```kt
/**
 * Some doc
 */
var violatingField: MutableList<String> = mutableListOf()
    set(requestHeaders) {
        field = requestHeaders.toMutableList()
    }
```

Detekt should report the line number as 4(1th based indexing)

## Observed Behavior
It reports the line number as 1

**Comment by BraisGabin** [comment]
We need to add tests for start and end report. We should report, if possible from `var` to the type, if any, or the name of there is not type.
